Stamina drugs & chroma effects

MissHeda opened the pull request:
adds:
- Pervitin (multistage drug with chroma & weapon sway effect)
- Caffeine (stamina refill drug)
- stamina boost for epi
- Chroma effect to Ketamine, Fentanyl & Pervitin

fix:
- Painkillers sound (was playing after animation)

MissHeda committed:
KAM + Drugs integration

add:
- Pervitin
- German translation (Pervitin)

fix:
- Painkillers sound (was playing after animation)

MiszczuZPolski commented on addons/pharma/XEH_PREP.hpp:
```diff
@@ -38,4 +38,6 @@ PREP(treatmentAdvanced_Reorientation);
 PREP(treatmentAdvanced_TXALocal);
-PREP(vehicleCheckDialysis);
\ No newline at end of file
+PREP(treatmentAdvanced_Pervitin);
```
it should be alphabetically :p

MiszczuZPolski commented on addons/pharma/functions/fnc_treatmentAdvanced_PervitinLocal.sqf:
```diff
@@ -0,0 +1,447 @@
+/// ACE Fatigue 
+
+if (ACEGVAR(advanced_fatigue,enabled)) then {
+	
```
remove one line here

MiszczuZPolski commented on addons/pharma/functions/fnc_treatmentAdvanced_PervitinLocal.sqf:
```diff
@@ -0,0 +1,447 @@
+},
+[_patient],
+10] call CBA_fnc_waitAndExecute;
+
```
remove one line

MiszczuZPolski commented on addons/pharma/functions/fnc_treatmentAdvanced_PervitinLocal.sqf:
```diff
@@ -0,0 +1,447 @@
+
+ /// ACE Fatigue Weapon Sway
+
+if (isNil "TAG_originalSwayFactor")    then {TAG_originalSwayFactor = ace_advanced_fatigue_swayFactor;};
```
change TAG to KAM

MiszczuZPolski commented on addons/pharma/functions/fnc_treatmentAdvanced_PervitinLocal.sqf:
```diff
@@ -0,0 +1,447 @@
+
+if (isNil "TAG_originalSwayFactor")    then {TAG_originalSwayFactor = ace_advanced_fatigue_swayFactor;};
+
+	[{
```
fix indents - so it looks better

MiszczuZPolski commented on addons/pharma/functions/fnc_treatmentAdvanced_PervitinLocal.sqf:
```diff
@@ -0,0 +1,447 @@
+
+/// Normal Stamina & Weapon Sway 
+
+if !(ACEGVAR(advanced_fatigue,enabled)) then {
```
if that "if" is just false of previous, use "else"

MiszczuZPolski commented on addons/pharma/functions/fnc_treatmentAdvanced_PervitinLocal.sqf:
```diff
@@ -0,0 +1,447 @@
+
+
+
+
```
leave only one line at the end

MiszczuZPolski commented on addons/pharma/ACE_Medical_Treatment.hpp:
```diff
@@ -189,5 +189,16 @@ class ADDON {
+            hrIncreaseHigh[] = {-10, 15};
+            timeInSystem = 600;
+            timeTillMaxEffect = 5;
+            maxDose = 4;
```
i would change it to 2 for balance purpose :v

MiszczuZPolski commented on addons/pharma/CfgWeapons.hpp:
```diff
@@ -20,6 +20,17 @@ class CfgWeapons {
     };
+    class kat_PervitinItem: ACE_ItemCore {
+        scope = 2; //This item is not actually used and it is replaced by ace_common_fnc_registerItemReplacement 
+        author = Miss Heda;
```
"Miss Heda"

MiszczuZPolski reviewed (changes requested):
quick look, once you apply changes to pervitin local, i will check it ingame

MissHeda committed:
Requested Changes

MissHeda committed:
Epinephrine Stamina boost

- Added a stamina boost for Epinephrine that is adaptive to normal and ace stamina.

MiszczuZPolski commented on addons/pharma/XEH_postInit.sqf:
```diff
@@ -18,4 +18,40 @@
+    params ["_medic", "_patient", "_bodyPart", "_classname"];
+    if (!local _patient) exitWith {["ace_treatmentSucceded", _this, _patient] call CBA_fnc_targetEvent};
+    if (_classname == "Epinephrine") then {
+        [_medic, _patient] call ace_medical_treatment_fnc_cprSuccess;
```
why this? if you copy things from ace discord at least check what you use

MissHeda replied:
Thought that is still needed for epi to run normally.

MiszczuZPolski commented on addons/pharma/XEH_postInit.sqf:
```diff
@@ -18,4 +18,40 @@
+    if (!local _patient) exitWith {["ace_treatmentSucceded", _this, _patient] call CBA_fnc_targetEvent};
+    if (_classname == "Epinephrine") then {
+        [_medic, _patient] call ace_medical_treatment_fnc_cprSuccess;
+        _patient setVariable [QGVAR(Epinephrine), 1, true];
```
where is it used

MiszczuZPolski commented on addons/pharma/XEH_postInit.sqf:
```diff
@@ -18,4 +18,40 @@
+
+		if (ACEGVAR(advanced_fatigue,enabled)) then {
+			params ["_patient"];
+			ace_advanced_fatigue_anReserve = ace_advanced_fatigue_anReserve + 1000;
```
i am not a fan of touching those gvars as Baer said but we will see in future

MissHeda replied:
I think baer ment changing things like the add-on settings mid game, like making the recovery factor better. This is only giving you a stamina reserve so I think it should be good.

MiszczuZPolski commented on addons/pharma/functions/fnc_treatmentAdvanced_PervitinLocal.sqf:
```diff
@@ -19,380 +20,332 @@ params ["_patient"];
 
+	/// ACE Fatigue Weapon Sway
 
+	if (isNil "KAM_originalSwayFactor")    then {
```
small note because i didnt check code that much yesterday -> instead of "KAM_originalSwayFactor" use QGVAR(originalSwayFactor)

MiszczuZPolski commented on addons/pharma/functions/fnc_treatmentAdvanced_PervitinLocal.sqf:
```diff
@@ -19,380 +20,332 @@ params ["_patient"];
-},
-[_patient],
-180] call CBA_fnc_waitAndExecute; /// 3m
+		[{
```
few codeblocks still need formatting "fix"

MissHeda committed:
requested changes

MissHeda committed:
requested changes

MiszczuZPolski committed:
Update fnc_treatmentAdvanced_PervitinLocal.sqf

MissHeda commented:
@MiszczuZPolski please hold on with pull, I had a great Idea and probably still need to add stuff to pervitin.

MissHeda committed:
Fixed code

MissHeda committed:
Chroma effect for Ketamine & Fentanyl

MissHeda committed:
Stamina "drugs" balancing changes

Also added Caffeine

MissHeda committed:
balancing changes

MiszczuZPolski commented on addons/pharma/functions/fnc_treatmentAdvanced_Caffeine.sqf:
```diff
@@ -0,0 +1,30 @@
+ * None
+ *
+ * Example:
+ * [player, cursorObject, "RightArm", "Pervitin", objNull, "kat_pervitin"] call kat_pharma_fnc_treatmentAdvanced_Pervitin;
```
caffeine, not pervitin

MiszczuZPolski commented on addons/pharma/functions/fnc_treatmentAdvanced_CaffeineLocal.sqf:
```diff
@@ -0,0 +1,38 @@
+#include "script_component.hpp"
+/*
+ * Author: Miss Heda
+ * 
```
remove line

MiszczuZPolski commented on addons/pharma/functions/fnc_treatmentAdvanced_CaffeineLocal.sqf:
```diff
@@ -0,0 +1,38 @@
+ * None
+ *
+ * Example:
+ * -
```
missing example

MiszczuZPolski commented on addons/pharma/functions/fnc_treatmentAdvanced_Fentanyl.sqf:
```diff
@@ -0,0 +1,29 @@
+#include "script_component.hpp"
+/*
+ * Author: MiszczuZPolski edited by Miss Heda
+ * Starts Lorazepam treatment
```
fentanyl, not lorazepam

MiszczuZPolski commented on addons/pharma/functions/fnc_treatmentAdvanced_Fentanyl.sqf:
```diff
@@ -0,0 +1,29 @@
+ * None
+ *
+ * Example:
+ * [player, cursorObject, "RightArm", "Lorazepam", objNull, "kat_lorazepam"] call kat_pharma_fnc_treatmentAdvanced_Lorazepam;
```
fix example

MiszczuZPolski commented on addons/pharma/functions/fnc_treatmentAdvanced_Ketamine.sqf:
```diff
@@ -0,0 +1,29 @@
+#include "script_component.hpp"
+/*
+ * Author: MiszczuZPolski edited by Miss Heda
+ * Starts Lorazepam treatment
```
ketamine

MiszczuZPolski commented on addons/pharma/functions/fnc_treatmentAdvanced_Ketamine.sqf:
```diff
@@ -0,0 +1,29 @@
+ * None
+ *
+ * Example:
+ * [player, cursorObject, "RightArm", "Lorazepam", objNull, "kat_lorazepam"] call kat_pharma_fnc_treatmentAdvanced_Lorazepam;
```
fix example

MiszczuZPolski commented on addons/pharma/functions/fnc_treatmentAdvanced_KetamineLocal.sqf:
```diff
@@ -0,0 +1,49 @@
+ * None
+ *
+ * Example:
+ * -
```
missing example

MiszczuZPolski commented on addons/pharma/functions/fnc_treatmentAdvanced_PervitinLocal.sqf:
```diff
@@ -0,0 +1,352 @@
+ * None
+ *
+ * Example:
+ * -
```
missing example

MiszczuZPolski commented on addons/pharma/ACE_Medical_Treatment.hpp:
```diff
@@ -189,5 +189,27 @@ class ADDON {
+            maxDose = 6;
+            incompatibleMedication[] = {};
+            viscosityChange = 0;
+        }
```
missing ;

MissHeda committed:
Requested changes

MissHeda committed:
Respawn fix

When you respawned the edited stamina & chroma effect was not set back to normal.

Fixed:
- stamina reset (ace & normal)
- remove chroma on respawn

Not fixed yet:
- script still runs & can change units stamina again

MissHeda committed:
Respawn fix 2

Now respawn removes:
- Weapon sway
- CamShake

MissHeda committed:
Pervitin 3D model + final respawn fix

MissHeda committed:
3D model (Caffeube)

MissHeda committed:
Everything working now

Gonna make some changes to the 3D model & icon in the future, but it can be pulled at it is now. :3

MissHeda committed:
3D model & image changes   [Ready to pull now}

MiszczuZPolski commented on addons/misc/functions/fnc_handleRespawn.sqf:
```diff
@@ -200,4 +200,46 @@ if (EGVAR(pharma,coagulation)) then {
+    
+    ["PDF"] call ace_advanced_fatigue_fnc_removeDutyFactor;
+	["EDF"] call ace_advanced_fatigue_fnc_removeDutyFactor;
+    ace_advanced_fatigue_swayFactor = GVAR(originalSwayFactor);
```
this GVAR should point to pharma

MiszczuZPolski commented on addons/misc/functions/fnc_handleRespawn.sqf:
```diff
@@ -200,4 +200,46 @@ if (EGVAR(pharma,coagulation)) then {
+
+} else {
+
+    params ["_patient"];
```
no need for params here

MiszczuZPolski commented on addons/misc/functions/fnc_handleRespawn.sqf:
```diff
@@ -200,4 +200,46 @@ if (EGVAR(pharma,coagulation)) then {
+    ppEffectDestroy _handle;
+};
+
+/// Terminate script
```
unused

MiszczuZPolski commented on addons/pharma/functions/fnc_treatmentAdvanced_Caffeine.sqf:
```diff
@@ -0,0 +1,37 @@
+
+if (ACE_Player != _patient) then {
+
+	[LLSTRING(NoSelfTreatment), 2, _medic] call ACEFUNC(common,displayTextStructured);
```
why this tho? i understand it

MissHeda replied:
Cause I want the player to know why it won't work if they give it to someone. Otherwise they'll report that as a bug.

MiszczuZPolski replied:
sounds kinda "stupid" that trooper cant take caffeine or pervitin by himself

MiszczuZPolski commented on addons/pharma/functions/fnc_treatmentAdvanced_FentanylLocal.sqf:
```diff
@@ -0,0 +1,50 @@
+					_handle ppEffectEnable false;
+					ppEffectDestroy _handle;
+				},
+				[_handle, _patient]
```
why passing patient?

MiszczuZPolski commented on addons/pharma/functions/fnc_treatmentAdvanced_KetamineLocal.sqf:
```diff
@@ -0,0 +1,50 @@
+					_handle ppEffectEnable false;
+					ppEffectDestroy _handle;
+				},
+				[_handle, _patient]
```
why passing patient?

MiszczuZPolski commented on addons/pharma/functions/fnc_treatmentAdvanced_Pervitin.sqf:
```diff
@@ -0,0 +1,37 @@
+
+if (ACE_Player != _patient) then {
+
+	[LLSTRING(NoSelfTreatment), 2, _medic] call ACEFUNC(common,displayTextStructured);
```
why this tho?

MiszczuZPolski reviewed (changes requested):
few things to correct ;v

MissHeda committed:
Requested changes <3

MissHeda committed:
mini change :3

MissHeda committed:
upsi :3

MissHeda committed:
Chroma effect turn off option

Adds the option to turn of chroma effect & weapon sway
Changes HR values for Pervitin & adjusts chroma effect for Pervitn, ketamine and Fentanyl

MissHeda committed:
Chromatic Aberration Sliders

MissHeda committed:
Fixed all known bugs

MissHeda committed:
Pharma settings fix & normal stamina adjustments

The IV / IO drop time text was to long to be displayed, added a new shorter text & moved the old long one to Desc of the new test.

Also adjusted the times for normal stamina.

MissHeda committed:
Epi stamina fix

Epi was only giving stamina effect if you admired it to yourself. Now it as well works if you are giving it to someone else.

MissHeda committed:
Merge branch 'Tomcat-SG:dev-Tomcat' into dev-Tomcat

MissHeda committed:
Merge branch 'Tomcat-SG:dev-Tomcat' into dev-Tomcat

MissHeda committed:
Painkiller & Vacuum sound updated

- made all painkiller sound calls being called with a painkillerStart to make it a 3D sound
- updated the Vacuum sound to be more realistic

MissHeda committed:
requested changes

AtrixZockt commented on addons/misc/functions/fnc_handleRespawn.sqf:
```diff
@@ -200,4 +200,42 @@ if (EGVAR(pharma,coagulation)) then {
+if (ACEGVAR(advanced_fatigue,enabled)) then {
+    
+    ["PDF"] call ace_advanced_fatigue_fnc_removeDutyFactor;
+	["EDF"] call ace_advanced_fatigue_fnc_removeDutyFactor;
```
format

AtrixZockt commented on addons/misc/functions/fnc_handleRespawn.sqf:
```diff
@@ -200,4 +200,42 @@ if (EGVAR(pharma,coagulation)) then {
+} else {
+
+    _patient enableStamina true;
+	_patient setAnimSpeedCoef 1;
```
format

AtrixZockt commented on addons/misc/functions/fnc_handleRespawn.sqf:
```diff
@@ -200,4 +200,42 @@ if (EGVAR(pharma,coagulation)) then {
+    _handle ppEffectAdjust _effect;
+    _handle ppEffectCommit 0;
+    addCamShake[0, 0, 50];
+    waitUntil{ppEffectCommitted _handle};
```
CBA_fnc_waitUntilandExecute

AtrixZockt commented on addons/pharma/XEH_preInit.sqf:
```diff
@@ -252,6 +252,53 @@ PREP_RECOMPILE_END;
+    "LIST",
+    [LLSTRING(medLvl_Pervitin)],
+    [CBA_SETTINGS_CAT, LSTRING(SubCategory_Pervitin)],
+    [[0, 1, 2], ["STR_ACE_Medical_Treatment_Anyone", "STR_ACE_Medical_Treatment_Medics", "STR_ACE_Medical_Treatment_Doctors"], 0],
```
[ACELSTRING (medical_treatment, Anyone), ACELSTRING (medical_treatment, Medics), ACELSTRING (medical_treatment, Doctors)]

MiszczuZPolski replied:
here it should be ACELSTRING

AtrixZockt commented on addons/pharma/XEH_preInit.sqf:
```diff
@@ -423,6 +470,24 @@ PREP_RECOMPILE_END;
 ] call CBA_Settings_fnc_init;
 
+[
+	QGVAR(chromatic_aberration_checkbox_ketamine),
+	"CHECKBOX",
+	[LLSTRING(SETTING_Chromatic_Aberration_Checkbox_Ketamine), LLSTRING(SETTING_Chromatic_Aberration_Checkbox_Ketamine_DESC)],
+	[CBA_SETTINGS_CAT, LSTRING(SubCategory_Ketamine)],
+	[true],
+	true
```
Format (look how the others are sorted)

AtrixZockt commented on addons/pharma/XEH_preInit.sqf:
```diff
@@ -423,6 +470,24 @@ PREP_RECOMPILE_END;
+[
+    QGVAR(chromatic_aberration_slider_ketamine),
+    "SLIDER",
+	[LLSTRING(SETTING_Chromatic_Aberration_Slider_Ketamine), LLSTRING(SETTING_Chromatic_Aberration_Slider_Ketamine_DESC)],
```
format

AtrixZockt commented on addons/pharma/XEH_preInit.sqf:
```diff
@@ -442,6 +507,24 @@ PREP_RECOMPILE_END;
 ] call CBA_Settings_fnc_init;
 
+[
+	QGVAR(chromatic_aberration_checkbox_fentanyl),
+	"CHECKBOX",
+	[LLSTRING(SETTING_Chromatic_Aberration_Checkbox_Fentanyl), LLSTRING(SETTING_Chromatic_Aberration_Checkbox_Fentanyl_DESC)],
+	[CBA_SETTINGS_CAT, LSTRING(SubCategory_Fentanyl)],
+	[true],
+	true
```
format (one TAB too much again)

AtrixZockt commented on addons/pharma/XEH_preInit.sqf:
```diff
@@ -442,6 +507,24 @@ PREP_RECOMPILE_END;
+[
+    QGVAR(chromatic_aberration_slider_fentanyl),
+    "SLIDER",
+	[LLSTRING(SETTING_Chromatic_Aberration_Slider_Fentanyl), LLSTRING(SETTING_Chromatic_Aberration_Slider_Fentanyl_DESC)],
```
format

AtrixZockt commented on addons/pharma/functions/fnc_treatmentAdvanced_CaffeineLocal.sqf:
```diff
@@ -0,0 +1,38 @@
+
+	[{
+		params ["_patient"];
+		ace_advanced_fatigue_anReserve = ace_advanced_fatigue_anReserve + 1500;
```
ACEGVAR(advanced_fatigue,anReserve)

AtrixZockt commented on addons/pharma/functions/fnc_treatmentAdvanced_Caffeine.sqf:
```diff
@@ -0,0 +1,30 @@
+ * None
+ *
+ * Example:
+ * [player, cursorObject, "RightArm", "Pervitin", objNull, "kat_pervitin"] call kat_pharma_fnc_treatmentAdvanced_Caffeine;
```
Change Perevitin to Caffeine

AtrixZockt commented on addons/pharma/functions/fnc_treatmentAdvanced_PervitinLocal.sqf:
```diff
@@ -0,0 +1,395 @@
+		params ["_patient"];
+
+		if !(alive _patient) exitWith {};
+		ace_advanced_fatigue_anReserve = ace_advanced_fatigue_anReserve + 3000;
```
ACEGVAR(advanced_fatigue,anReserve)

AtrixZockt commented on addons/pharma/functions/fnc_treatmentAdvanced_PervitinLocal.sqf:
```diff
@@ -0,0 +1,395 @@
+		params ["_patient"];
+
+		if !(alive _patient) exitWith {};
+		ace_advanced_fatigue_anReserve = ace_advanced_fatigue_anReserve + 3000;
```
ACEGVAR(advanced_fatigue,anReserve)

AtrixZockt commented on addons/pharma/functions/fnc_treatmentAdvanced_PervitinLocal.sqf:
```diff
@@ -0,0 +1,395 @@
+		params ["_patient"];
+
+		if !(alive _patient) exitWith {};
+		ace_advanced_fatigue_anReserve = ace_advanced_fatigue_anReserve + 3000;
```
ACEGVAR(advanced_fatigue,anReserve)

AtrixZockt commented on addons/pharma/functions/fnc_treatmentAdvanced_PervitinLocal.sqf:
```diff
@@ -0,0 +1,395 @@
+		[{
+			params ["_patient"];
+			if !(alive _patient) exitWith {};
+			ace_advanced_fatigue_swayFactor = GVAR(originalSwayFactor) * 0.3;
```
ACEGVAR(advanced_fatigue, swayFactor)

AtrixZockt commented on addons/pharma/functions/fnc_treatmentAdvanced_PervitinLocal.sqf:
```diff
@@ -0,0 +1,395 @@
+		[{
+			params ["_patient"];
+			if !(alive _patient) exitWith {};
+			ace_advanced_fatigue_swayFactor = GVAR(originalSwayFactor) * 0.5 ;
```
ACEGVAR(advanced_fatigue, swayFactor)

AtrixZockt commented on addons/pharma/functions/fnc_treatmentAdvanced_PervitinLocal.sqf:
```diff
@@ -0,0 +1,395 @@
+		[{
+			params ["_patient"];
+			if !(alive _patient) exitWith {};
+			ace_advanced_fatigue_swayFactor = GVAR(originalSwayFactor) * 1;
```
ACEGVAR(advanced_fatigue, swayFactor)

AtrixZockt commented on addons/pharma/functions/fnc_treatmentAdvanced_PervitinLocal.sqf:
```diff
@@ -0,0 +1,395 @@
+		[{
+			params ["_patient"];
+			if !(alive _patient) exitWith {};
+			ace_advanced_fatigue_swayFactor = (GVAR(originalSwayFactor) * 1.3) min 2;
```
Im too lazy to mention all.  
Replace "ace_advanced_fatigue_swayFactor" with "ACEGVAR(advanced_fatigue, swayFactor)"

AtrixZockt reviewed (changes requested):
Other than the maked ones, looks good

MissHeda committed:
requested changes UwU

MissHeda committed:
Requested changes

AtrixZockt reviewed (approved):
Everything good and tested

MissHeda committed:
Pneumo. debuff

Adds a white flashings effect if you go below 90% SPO2 simulating losing your breath. (CBA setting of course)
Also removes a bit of stamina every 3 seconds.

MissHeda committed:
Pneuo debuff changes

Changed white to black

MissHeda committed:
Requested changes
